### ISSUE TITLE
Doesn't install properly on NixOS

### ISSUE THREAD

**Issue by NilsIrl** [glacambre/firenvim#365]
I am just opening this issue as a report. Do not waste your time trying to fix it unless you use NixOS.

Basically NixOS uses a wrapper for some programs which allows to set an environment (like setting up the providers). The problem is that the value of `v:progpath` is set to the unwrapped executable which means that `~/.local/share/firenvim/firenvim` doesn't set up the environment correctly.

I believe the best fix for this would be for `v:progpath` to be set to the wrapper which would be a NixOS issue.

Or maybe `v:progpath` isn't the variable `firenvim` should be using to determine where the nvim executable is.

Another solution would be to put inside `~/.local/share/firenvim/firenvim` the arguments that `nvim` was opened with, in the same way `v:progpath` is used to get the executable.

**Comment by glacambre** [comment]
Firenvim already makes some [environment-specific tests](https://github.com/glacambre/firenvim/blob/178b739bee139b7549dcd533dd8e62defd4c1b65/autoload/firenvim.vim#L221) so I wouldn't mind adding a NixOs-specific test. I don't use NixOs though so I don't want to do this myself. But feel free to open a PR if you figure out what needs to be done :)

Related issue: https://github.com/NixOS/nixpkgs/issues/77633

**Comment by NilsIrl** [comment]
I personally fixed it just by calling `nvim`

**Comment by glacambre** [comment]
The problem with doing that is that if Neovim isn't in your "system path" (= it's in a directory that you add to your $PATH on login/shell startup) and you use a desktop environment that doesn't source your login scripts (I think that's what Gnome does on Ubuntu by default), the `firenvim` shell script won't be able to start Neovim.

**Comment by glacambre** [comment]
@NilsIrl what's `v:progpath`'s value in a neovim installed through Nix? Are variables such as $NIX_PATH and $IN_NIX_SHELL defined?

**Comment by NilsIrl** [comment]
> what's `v:progpath`'s value in a neovim installed through Nix?

It is set to the unwrapped binary. This means firenvim will call the unwrapped binary and some stuff won't be set correctly (python and ruby stuff).

This is what the wrapper (which ideally firenvim should be calling) looks like:

```bash
#! /nix/store/xb062l4b76zyhq6grqf4iyfdikkpg8fl-bash-4.4-p23/bin/bash -e
export PATH=$PATH${PATH:+':'}'/nix/store/mcmcfvfmal42vd9pafhcgxjpyivq8ycy-neovim-ruby-env/bin'
export GEM_HOME='/nix/store/mcmcfvfmal42vd9pafhcgxjpyivq8ycy-neovim-ruby-env/lib/ruby/gems/2.6.0'
exec "/nix/store/p3s6iv851x0857ys7zngz06ij9fghbay-neovim-unwrapped-0.4.3/bin/nvim"   --cmd "let g:loaded_node_provider=1" --cmd "let g:python_host_prog='/nix/store/km8a43dvpfjijb5rb1xq3b3j2k37yk0n-neovim-0.4.3/bin/nvim-python'" --cmd "let g:python3_host_prog='/nix/store/km8a43dvpfjijb5rb1xq3b3j2k37yk0n-neovim-0.4.3/bin/nvim-python3'" --cmd "let g:ruby_host_prog='/nix/store/km8a43dvpfjijb5rb1xq3b3j2k37yk0n-neovim-0.4.3/bin/nvim-ruby'"  "$@"
```

**Comment by glacambre** [comment]
Thanks. Is `g:python_host_prog` correctly set if you run neovim from your shell?

**Comment by NilsIrl** [comment]
> Thanks. Is `g:python_host_prog` correctly set if you run neovim from your shell?

yes. Now that I've changed `~/.local/share/firenvim/firenvim`, it works inside firenvim as well.

**Comment by glacambre** [comment]
Closed in 6cd421e, thanks for answering my questions :)
